lib/chef/provider/service/systemd.rb

```diff
@@ -76,6 +76,30 @@ def define_resource_requirements
     end
   end
 
+  def systemd_service_status
+    @service_status ||= begin
+      # Collect all the status information for a service and returns it at once
+      options, args = get_systemctl_options_args
+      s = shell_out!("#{systemctl_path} #{args} show -p UnitFileState -p ActiveState #{new_resource.service_name}", options)
```

Comment:
you should use the *splat args style of calling here so:

```
shell_out!(systemctl_path,  args,  "show", "-p", "UnitFileState", "-p", "ActiveState", new_resource.service_name, options)
```

if that gets done correctly then it should avoid invoking the shell to parse the arguments, which should increase the perf even more (and has other benefits in not needing to worry about filenames with spaces and quotes around them, etc, etc).